### ISSUE TITLE
Why use 'torch.nn.Bilinear'  in 'self.channelMixMLPs01' ?

### ISSUE THREAD

**Issue by linhaojia13** [LifeBeyondExpectations/ECCV22-PointMixer#9]
What's the purpose of using 'torch.nn.Bilinear'? 
The formulation of a bilinear transformation is y= x_1^T A x_2 + b, and the formulation of a linear transformation is y=xA^T+b .
It seems that a bilinear layer just apply a slighter sophisticated linear transformation than the linear layer?

**Comment by LifeBeyondExpectations** [comment]
We empirically found that `nn.Bilinear(...)` results in a good performance. 

Nonetheless, in my personal opinion, this function can be seen as the 'simplest' version of a conditional MLP.

https://github.com/LifeBeyondExpectations/ECCV22-PointMixer/blob/bf1c3fb7187d737d68901e7165e78d64ab3abe4d/sem_seg/model/network/pointmixer.py#L29-L38

As you can see the code above, the bilinear function takes the same input x (line 37). 
Accordingly, we can say that 
<pre>
y = (x<sup>T</sup> W) x
  = W' x 
</pre>
where `W'` is conditioned to the input x.

In short, I think that not that much 'purpose' exists in this function.
However, `nn.Bilinear(...)` can be viewed as another type of MLP, as its name implies.

Hope you satisfied with my understanding.

**Comment by linhaojia13** [comment]
Thank you for your  detailed explanation. I think it is reasonable that regarding the bilinear layer as a simplest implementation of conditional MLP. 
I'm looking forward to seeing the results of ablation study. Thank you!
